webapp/views/wpt-app.js

```diff
@@ -235,8 +242,45 @@ class WPTApp extends PathInfo(WPTFlags(TestRunsUIBase)) {
     };
     this.handleCollapse = () => {
       this.isBSFCollapsed = !this.isBSFCollapsed;
+      // Record hide/open actions on the BSF graph. Currently, we only
+      // show it on the homepage.
+      if ('ga' in window) {
+        window.ga('send', {
+          hitType: 'event',
+          eventCategory: 'bsf',
+          eventAction: 'hide',
+          eventLabel: this.path,
+          eventValue: this.isBSFCollapsed ? 1 : 0
+        });
+      }
       this.setLocalStorageFlag(this.isBSFCollapsed, 'isBSFCollapsed');
     };
+    this.enterBSF = () => {
+      if (this.isInteracting) {
+        return;
+      }
+      this.bsfStartTime = new Date();
+    };
+    this.exitBSF = () => {
+      if (this.isInteracting || !this.bsfStartTime) {
```

Comment:
`this.bsfStartTime` is never cleared, so after the first call to `exitBSF` it will always be non-null and won't trigger this early return (which I presume is meant to guard against `exitBSF` firing with an equivalent `enterBSF`?)

Reply:
Yea it is somewhat redundant. I put it here just in case weird things happen. Happy to remove it if it confuses you

Reply:
This is fine, but we should probably clear `bsfStartTime` in `exitBSF` after using it then.